entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/config/EntityUpdateJobFactory.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+import java.util.List;
+
+import static eu.europeana.entitymanagement.batch.utils.BatchUtils.*;
+import static eu.europeana.entitymanagement.common.vocabulary.AppConfigConstants.*;
+
+/**
+ * Entity Job update factory class
+ * @author srishti singh
+ * @since 14 July 2025
+ */
+@Component
+public class EntityUpdateJobFactory {
+
+    private final JobBuilderFactory jobBuilderFactory;
+    private final StepBuilderFactory stepBuilderFactory;
+    private final ScheduledTaskItemListener itemListener;
+    private final ScheduledTaskService scheduledTaskService;
+
+    /** SkipPolicy to ignore all failures when executing jobs, as they can be handled later */
+    private final SkipPolicy noopSkipPolicy = (Throwable t, int skipCount) -> true;
+
+    @Resource
+    EntityManagementConfiguration emConfig;
+
+    @Resource
+    AppAutoconfig emAutoConfig;
+
+    public EntityUpdateJobFactory(JobBuilderFactory jobBuilderFactory, StepBuilderFactory stepBuilderFactory,
+                                  ScheduledTaskItemListener itemListener,
+                                  ScheduledTaskService scheduledTaskService) {
+        this.jobBuilderFactory = jobBuilderFactory;
+        this.stepBuilderFactory = stepBuilderFactory;
+        this.itemListener = itemListener;
+        this.scheduledTaskService = scheduledTaskService;
+    }
+
+    /**
+     * Creates Job via JobBuilderFactory.
+     * Expects `entityId` string in JobParameters. This would
+     * typically be run synchronously
+     */
+    public Job createJob(List<Task> processors, List<Task> writers) {
+        return this.jobBuilderFactory
+                .get(JOB_UPDATE_SINGLE_ENTITY)
+                .incrementer(new RunIdIncrementer())
+                // this job is always launched from web requests, so synchronousTaskExecutor is used. It
+                // also directly retrieves entities from the EntityRecord database.
+                .start(synchronousUpdate(processors, writers))
+                .build();
+    }
+
+    /**
+     * Creates a step for synchronous update
+     * @return
+     */
+    private Step synchronousUpdate(List<Task> processors, List<Task> writers) {
+        return this.stepBuilderFactory
+                .get(STEP_UPDATE_ENTITY)
+                .<BatchEntityRecord, BatchEntityRecord>chunk(1)
+                .reader(getReader(true))
+                .processor(getProcessor(processors))
+                .writer(getWriter(writers))
+                .listener((ItemProcessListener<? super BatchEntityRecord, ? super BatchEntityRecord>) itemListener)
+                .faultTolerant()
+                .skipPolicy(noopSkipPolicy)
+                .taskExecutor((TaskExecutor)getApplicationContext().getBean(WEB_REQUEST_JOB_EXECUTOR))
```

Comment:
better move to a getter method, as it is likely to reuse that method in more places

Reply:
done